tests/tracking/test_tracking.py

```diff
@@ -722,23 +740,28 @@ def test_get_artifact_uri_uses_currently_active_run_id():
             "mysql+driver://user:password@host:port/dbname/subpath/{run_id}/artifacts/{path}#fragment",  # pylint: disable=line-too-long
         ),
         ("s3://bucketname/rootpath", "s3://bucketname/rootpath/{run_id}/artifacts/{path}"),
-        ("/dirname/rootpa#th?", "/dirname/rootpa#th?/{run_id}/artifacts/{path}"),
     ],
 )
 def test_get_artifact_uri_appends_to_uri_path_component_correctly(
     artifact_location, expected_uri_format
 ):
-    client = MlflowClient()
-    client.create_experiment("get-artifact-uri-test", artifact_location=artifact_location)
-    mlflow.set_experiment("get-artifact-uri-test")
-    with mlflow.start_run():
-        run_id = mlflow.active_run().info.run_id
-        for artifact_path in ["path/to/artifact", "/artifact/path", "arty.txt"]:
-            artifact_uri = mlflow.get_artifact_uri(artifact_path)
-            assert artifact_uri == tracking.artifact_utils.get_artifact_uri(run_id, artifact_path)
-            assert artifact_uri == expected_uri_format.format(
-                run_id=run_id, path=artifact_path.lstrip("/")
-            )
+    _assert_get_artifact_uri_appends_to_uri_path_component_correctly(
+        artifact_location, expected_uri_format
+    )
+
+
+@pytest.mark.skipif(not is_local_os_windows(), reason="This test only passes on Windows")
+def test_get_artifact_uri_appends_to_local_path_component_correctly_on_windows():
+    _assert_get_artifact_uri_appends_to_uri_path_component_correctly(
+        "/dirname/rootpa#th?", "file:///{drive}/dirname/rootpa/{run_id}/artifacts/{path}#th?"
+    )
+
+
+@pytest.mark.skipif(is_local_os_windows(), reason="This test fails on Windows")
+def test_get_artifact_uri_appends_to_local_path_component_correctly():
+    _assert_get_artifact_uri_appends_to_uri_path_component_correctly(
+        "/dirname/rootpa#th?", "{drive}/dirname/rootpa#th?/{run_id}/artifacts/{path}"
+    )
```

Comment:
So on windows, `"/dirname/rootpa#th?"` is resolved to a URI, on non-windows OS, `"/dirname/rootpa#th?"` is resolved to a path?

Reply:
yes, that is correct! that is the intended change in this PR.